RobotCode/snobot2018/src/org/snobot/elevator/SnobotElevator.java

```diff
@@ -85,44 +108,87 @@ public void initializeLogHeaders()
     @Override
     public void updateLog()
     {
-        mLogger.updateLogger(mHeight);
-        mLogger.updateLogger(mElevatorSpeed);
+        mLogger.updateLogger(mActualHeight);
+        mLogger.updateLogger(mJoystickSpeed);
 
     }
 
     @Override
     public void updateSmartDashboard()
     {
-        SmartDashboard.putNumber(SmartDashboardNames.sELEVATOR_HEIGHT, mHeight);
-        SmartDashboard.putNumber(SmartDashboardNames.sELEVATOR_MOTOR_SPEED, mElevatorSpeed);
+        SmartDashboard.putNumber(SmartDashboardNames.sELEVATOR_HEIGHT, mActualHeight);
+        SmartDashboard.putNumber(SmartDashboardNames.sELEVATOR_MOTOR_SPEED, mJoystickSpeed);
 
     }
 
     @Override
     public double getHeight()
     {
 
-        return mHeight;
+        return mActualHeight;
     }
 
     @Override
     public void setMotorSpeed(double aSpeed)
     {
-        mElevatorMotor.set(aSpeed);
+        if (canMove(aSpeed))
+        {
+            mElevatorMotor.set(aSpeed);
+        }
+        else
+        {
+            stop();
+        }
+    }
 
+    /**
+     * Decides if the elevator can or can't move.
+     * 
+     * @param aDirection
+     *            is the direction the elevator wants to move.
+     * @return If true the elevator can move and if false then the elevator
+     *         can't move.
+     */
+    protected boolean canMove(double aDirection)
+    {
+        boolean canMove = true;
+
+        if (aDirection > 0 && (mMaxHeight <= mActualHeight))
+        {   
+            canMove = false;
+        }
+        else if (aDirection < 0 && (mMinHeight >= mActualHeight))
+        {
+            canMove = false;
+        }
+        
+        return canMove;
     }
 
     @Override
-    public void goUp()
+    public void setHeight(double aHeight)
```

Comment:
Is there a reason why there is are separate `goToHeight` and `setHeight` functions?  I don't immediately see a need and it can cause state confusion.  Combining them into one function (`goToHeight(double aHeight`) could make it clearer what is going on

Reply:
The goToHeight function is used in autonomous mode.  Not sure if there was a better way but if autonomous doesn't have a control function we needed to do something.